cpp/lib.py

```diff
@@ -871,29 +871,18 @@ def merge_clang_tidy_checks(cls, orig_checks, new_checks):
         new_checks = [check.strip() for check in new_checks.split(",")]
 
         for new_check in new_checks:
-            if new_check.startswith("-"):
-                name = new_check[1:]
-                # remove check when check=google-runtime-references to_=-google-*
-                orig_checks = list(
-                    check for check in orig_checks if not fnmatch(check, name)
-                )
-                # remove check when check=-google-runtime-references
-                # to_=-google-* (simplification)
-                orig_checks = list(
-                    check for check in orig_checks if not fnmatch(check, new_check)
-                )
-            else:
-                # remove check when check=-google-runtime-references to_=google-*
-                orig_checks = list(
-                    check
-                    for check in orig_checks
-                    if not fnmatch(check, "-" + new_check)
-                )
-                # remove check when check=google-runtime-references
-                # to_=google-* (simplification)
-                orig_checks = list(
-                    check for check in orig_checks if not fnmatch(check, new_check)
-                )
+            name_without_prefix = (
+                new_check[1:] if new_check.startswith("-") else new_check
+            )
+            name_with_prefix = "-" + name_without_prefix
+            orig_checks = list(
+                check for check in orig_checks if not fnmatch(check, name_with_prefix)
+            )
+            orig_checks = list(
+                check
+                for check in orig_checks
+                if not fnmatch(check, name_without_prefix)
+            )
```

Comment:
There's two ways of seeing what's happening:
1. First we remove everything that doesn't match `check -> fnmatch(check, name_with_prefix)`. Next we remove everything that doesn't match `check -> fnmatch(check, name_with_prefix)`.
2. Keep everything that matches `check -> not fnmatch(check, name_with_prefix) and not fnmatch(check, name_without_prefix)`.

The second we could refactor as:
```
def keep_if(check):
    matches_with_prefix = fnmatch(check, name_with_prefix)
    matches_without_prefix =  fnmatch(check, name_with_prefix)

    return not matches_with_prefix and not matches_without_prefix

# Or list(filter(...)) 
orig_checks = [check for check in origin_checks if keep_if(check)]
```

Why do I prefer the second option? I don't need to check two list comprehensions, I feel like the parts are broken down into smaller logical blocks, and I don't need to figure out if the order matters. That said, this is purely a personal preference.

Reply:
Thank you for the feedback. I will keep the current logic.